cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/LoadMonitor.java

```diff
@@ -500,6 +547,43 @@ public ClusterModel clusterModel(long from,
     return clusterModel;
   }
 
+  /**
+   * Get replica placement information, i.e. each replica resides on which disk of the broker.
+   *
+   * @param clusterModel The cluster model to populate replica placement information.
+   * @param cluster The cluster metadata.
+   * @return A map from topic partition to replica placement information.
+   *
+   */
+  private Map<TopicPartition, Map<Integer, String>> getReplicaPlacementInfo(ClusterModel clusterModel, Cluster cluster) {
+    Map<TopicPartition, Map<Integer, String>> replicaPlacementInfo = new HashMap<>();
+    Map<Integer, KafkaFuture<Map<String, LogDirInfo>>> logDirsByBrokerId =
+        _adminClient.describeLogDirs(cluster.nodes().stream().mapToInt(Node::id).boxed().collect(Collectors.toList())).values();
+    for (Map.Entry<Integer, KafkaFuture<Map<String, LogDirInfo>>> entry : logDirsByBrokerId.entrySet()) {
+      Integer brokerId =  entry.getKey();
+      try {
+        entry.getValue().get(LOGDIR_RESPONSE_TIMEOUT_MS, TimeUnit.MILLISECONDS).forEach((logdir, info) -> {
+          if (info.error == Errors.NONE) {
+            for (Map.Entry<TopicPartition, ReplicaInfo> e : info.replicaInfos.entrySet()) {
+              if (!e.getValue().isFuture) {
+                replicaPlacementInfo.putIfAbsent(e.getKey(), new HashMap<>());
+                replicaPlacementInfo.get(e.getKey()).put(brokerId, logdir);
```

Comment:
Can we avoid additional `get` call? -- i.e. `replicaPlacementInfo.putIfAbsent(e.getKey(), new HashMap<>()).put(brokerId, logdir);`

Reply:
IIUC,  if the key does not exist, `putIfAbsent` will return `null`. 

> default V putIfAbsent(K key,
                      V value)
If the specified key is not already associated with a value (or is mapped to null) associates it with the given value and returns null, else returns the current value.

Reply:
Ah right, I was thinking about `getOrDefault`, you are right!